dask/dataframe/rolling.py

```diff
@@ -34,50 +33,106 @@ def overlap_chunk(
 
     parts = [p for p in (prev_part, current_part, next_part) if p is not None]
     combined = methods.concat(parts)
-    out = func(combined, *args, **kwargs)
-    if prev_part is None:
+
+    return CombinedOutput(
+        (
+            combined,
+            len(prev_part) if prev_part is not None else None,
+            len(next_part) if next_part is not None else None,
+        )
+    )
+
+
+def overlap_chunk(func, before, after, *args, **kwargs):
+    dfs = [df for df in args if isinstance(df, CombinedOutput)]
+    combined, prev_part_length, next_part_length = dfs[0]
+
+    args = [arg[0] if isinstance(arg, CombinedOutput) else arg for arg in args]
+
+    out = func(*args, **kwargs)
+
+    if prev_part_length is None:
         before = None
     if isinstance(before, datetime.timedelta):
-        before = len(prev_part)
+        before = prev_part_length
 
     expansion = None
     if combined.shape[0] != 0:
         expansion = out.shape[0] // combined.shape[0]
     if before and expansion:
         before *= expansion
-    if next_part is None:
+    if next_part_length is None:
         return out.iloc[before:]
     if isinstance(after, datetime.timedelta):
-        after = len(next_part)
+        after = next_part_length
     if after and expansion:
         after *= expansion
     return out.iloc[before:-after]
 
 
-def map_overlap(func, df, before, after, *args, **kwargs):
+def map_overlap(
+    func,
+    before,
+    after,
+    *args,
+    meta=no_default,
+    enforce_metadata=True,
+    transform_divisions=True,
+    align_dataframes=True,
+    **kwargs,
+):
     """Apply a function to each partition, sharing rows with adjacent partitions.
 
+    Note that the index and divisions are assumed to remain unchanged.
```

Comment:
`transform_divisions`, which defaults to `True, suggests the index+divisions change, right?

Reply:
I see this note is also in [`map_partitions`](https://docs.dask.org/en/stable/generated/dask.dataframe.DataFrame.map_partitions.html) -- which might also need updating

Reply:
Yes, for almost all parameters (except before and after) the documentation is equal to map_patitions.

Reply:
Thanks! This note does need to be removed in both the docstrings. Feel free to include this in your PR, we can also do it separately. :)